cmd/catalyst-node/catalyst-node.go

```diff
@@ -424,18 +424,58 @@ func writeSerfConfig(config *agent.Config) (string, error) {
 }
 
 func startCatalystWebServer(httpAddr string, redirectPrefixes []string) {
-	http.Handle("/hls/", redirectHlsHandler(redirectPrefixes))
+	http.Handle("/", redirectHandler(redirectPrefixes))
 	glog.Infof("HTTP server listening on %s", httpAddr)
 	glog.Fatal(http.ListenAndServe(httpAddr, nil))
 }
 
 var getClosestNode = queryMistForClosestNode
 
-func redirectHlsHandler(redirectPrefixes []string) http.Handler {
+// return the best node available for a given stream. will return any node if nobody has the stream.
+func getBestNode(redirectPrefixes []string, playbackID, lat, lon string) (string, string, error) {
+	var nodeAddr, fullPlaybackID, fallbackAddr string
+	var err error
+	var waitGroup sync.WaitGroup
+
+	for _, prefix := range redirectPrefixes {
+		waitGroup.Add(1)
+		go func(prefix string) {
+			addr, e := getClosestNode(playbackID, lat, lon, prefix)
+			if e != nil {
+				err := e
+				glog.V(8).Infof("error finding origin server playbackID=%s prefix=%s error=%s", playbackID, prefix, err)
+				// If we didn't find a stream but we did find a server, keep that so we can use it to handle a 404
+				if addr != "" {
```

Comment:
I added a similar comment somewhere, but I think this logic looks fishy. I think here is the point where we should query for the fallback node (meaning for any node). 

However we query for ANY node in `getCloseNode` which seems very confusing.

Reply:
Hmm, it might be just be the case that we have the wrong prefix though; we can't be sure there's not a solid match for the stream elsewhere. I agree this logic is confusing/feels messy. I think the right way for us to do this wouldn't be to query Mist at all, and instead use our own list of servers to return a fallback when appropriate. But we don't yet have any geolocation logic or anything outside of MistUtilLoad, so it's tricky.